src/internal_events/docker_logs.rs

```diff
@@ -104,77 +124,116 @@ impl<'a> InternalEvent for DockerLogsCommunicationError<'a> {
         error!(
             message = "Error in communication with Docker daemon.",
             error = ?self.error,
+            error_type = "connection_failed",
+            stage = "receiving",
             container_id = ?self.container_id,
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => self.error.to_string(),
+            "error_type" => "connection_failed",
+            "stage" => "receiving",
+        );
+        // deprecated
         counter!("communication_errors_total", 1);
     }
 }
 
 #[derive(Debug)]
-pub struct DockerLogsContainerMetadataFetchFailed<'a> {
+pub struct DockerLogsContainerMetadataFetchError<'a> {
     pub error: Error,
     pub container_id: &'a str,
 }
 
-impl<'a> InternalEvent for DockerLogsContainerMetadataFetchFailed<'a> {
+impl<'a> InternalEvent for DockerLogsContainerMetadataFetchError<'a> {
     fn emit_logs(&self) {
         error!(
             message = "Failed to fetch container metadata.",
             error = ?self.error,
+            error_type = "request_failed",
+            stage = "receiving",
             container_id = ?self.container_id,
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => self.error.to_string(),
+            "error_type" => "request_failed",
+            "stage" => "receiving",
+            "container_id" => self.container_id.to_owned(),
+        );
+        // deprecated
         counter!("container_metadata_fetch_errors_total", 1);
     }
 }
 
 #[derive(Debug)]
-pub struct DockerLogsTimestampParseFailed<'a> {
+pub struct DockerLogsTimestampParseError<'a> {
     pub error: ParseError,
     pub container_id: &'a str,
 }
 
-impl<'a> InternalEvent for DockerLogsTimestampParseFailed<'a> {
+impl<'a> InternalEvent for DockerLogsTimestampParseError<'a> {
     fn emit_logs(&self) {
         error!(
             message = "Failed to parse timestamp as RFC3339 timestamp.",
             error = ?self.error,
+            error_type = "parser_failed",
+            stage = "processing",
             container_id = ?self.container_id,
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "error" => self.error.to_string(),
+            "error_type" => "parser_failed",
```

Comment:
We don't need to do this now, but pulling this up into some enum would probably increase the chances that we use a consistent error type and stage across components.